### PR TITLE
Fix std::bad_alloc() in WaveformRenderBeat::draw()

### PR DESCRIPTION
**Pull request by uklotzde**
https://bugs.launchpad.net/mixxx/+bug/1977662
https://bugs.launchpad.net/mixxx/+bug/1959489

@ronso: Please test if this fixes your issues. Then we could merge it instead of #4787.

### CODE REVIEW

File: src/engine/enginebuffer.cpp [hunk 1]
```diff
@@ -1242,11 +1242,12 @@ void EngineBuffer::postProcess(const int iBufferSize) {
 }
 
 void EngineBuffer::updateIndicators(double speed, int iBufferSize) {
-    if (m_trackSampleRateOld == 0) {
+    if (m_filepos_play == kInitialSamplePosition || m_trackSampleRateOld == 0) {
```
Comment (Holzhaus) [inline src/engine/enginebuffer.cpp:1245]: use of undeclared identifier 'kInitialSamplePosition'; did you mean 'kInitalSamplePosition'?

**Comment by ronso0** [comment]
Hehe, build fails because of a typo (in the source, not your code ; )

**Comment by ronso0** [comment]
Yes, thank you, that also fixes the crash.
Also, having a blank waveform is consistent with the state after
* Mixxx started, no track loaded
* activate passthrough
* load track

compared to the solution in #4787.

Though, the playposition in the overview is still stuck at the center.
I think I'll fix this in the woverview code.

**Comment by uklotzde** [comment]
Ouch, there is a typo. I tested only the `main` version.

**Review by ronso0 — CHANGES_REQUESTED** [review]

**Comment by ronso0** [comment]
Thank you very much, LGTM

**Review by ronso0 — APPROVED** [review]

**Comment by uklotzde** [comment]
Merged into `main`.

**Comment by ronso0** [comment]
thanks, I got interrupted by real life and was about to do it just now.

**Comment by uklotzde** [comment]
The new debug assertions unexpectedly triggers on main:

```
critical [Engine] DEBUG ASSERT: "m_tempo_ratio_old != 0" in function void EngineBuffer::updateIndicators(double, int) at /home/uk/volumes/Build/cpp/mixxx/src/engine/enginebuffer.cpp:1349

 72/786 Test #708: SoundSourceProxyTest.skipAndRead ...........................................................   Passed   59.04 sec
```

**Pull request merged**